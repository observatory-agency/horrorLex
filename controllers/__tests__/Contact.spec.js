const ContactController = require('../Contact');

describe('ContactController', () => {
  let contact;
  beforeEach(() => {
    contact = new ContactController();
  });

  describe('constructor', () => {
    it('should set its templates', () => {
      expect(contact.template).toMatchSnapshot();
    });
  });

  describe('get', () => {
    it('should call render with the "get" template', () => {
      const renderMock = jest.fn();
      const resMock = { render: renderMock };
      contact.get(null, resMock);
      expect(renderMock).toHaveBeenCalledWith(contact.template.get);
    });
  });
});
