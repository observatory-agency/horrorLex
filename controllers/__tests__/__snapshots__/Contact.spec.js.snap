// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactController constructor should set its templates 1`] = `
Object {
  "get": "contact.hbs",
}
`;
